cpp/models/abm/person.h

```diff
@@ -385,52 +386,42 @@ class Person
     ScalarType get_mask_protective_factor(const Parameters& params) const;
 
     /**
-     * @brief For every #LocationType a Person has a compliance value between -1 and 1.
-     * -1 means that the Person never complies to any Mask duty at the given #LocationType.
-     * 1 means that the Person always wears a Mask a the #LocationType even if it is not required.
-     * @param[in] preferences The vector of Mask compliance values for all #LocationType%s.
+     * @brief For every #InterventionType a Person has a compliance value between 0 and 1.
+     * 0 means that the Person never complies to the Intervention.
+     * 1 means that the Person always complies to the Intervention.
+     * @param[in] intervention_type The intervention type. 
+     * @param[in] value The compliance value.
      */
-    void set_mask_preferences(std::vector<ScalarType> preferences)
+    void set_compliance(InterventionType intervention_type, ScalarType value)
     {
-        m_mask_compliance = preferences;
+        m_compliance[static_cast<uint32_t>(intervention_type)] = value;
     }
 
     /**
-     * @brief Get the Mask compliance of the Person for the current Location.
-     * @param[in] location The current Location of the Person.
-     * @return The probability that the Person does not comply to any Mask duty/wears a Mask even if it is not required.
+     * @brief Get the compliance of the Person for an intervention.
+     * @param[in] intervention_type The intervention type. 
+     * @return The probability that the Person complies to an intervention.
      */
-    ScalarType get_mask_compliance(LocationType location) const
+    ScalarType get_compliance(InterventionType intervention_type) const
     {
-        return m_mask_compliance[static_cast<int>(location)];
+        return m_compliance[static_cast<uint32_t>(intervention_type)];
     }
 
     /**
-     * @brief Checks whether the Person wears a Mask at the target Location.
+     * @brief Checks whether the Person is following an intervention.
      * @param[inout] rng RandomNumberGenerator of the Person.
-     * @param[in] target The target Location.
-     * @return Whether a Person wears a Mask at the Location.
-     */
-    bool apply_mask_intervention(RandomNumberGenerator& rng, const Location& target);
-
-    /**
-     * @brief Decide if a Person is currently wearing a Mask.
-     * @param[in] wear_mask If true, the protection of the Mask is considered when computing the exposure rate.
+     * @param[in] intervention The InterventionType.
+     * @return Checks whether the Person following an intervention.
      */
-    void set_wear_mask(bool wear_mask)
-    {
-        m_wears_mask = wear_mask;
-    }
+    bool is_compliant(RandomNumberGenerator& rng, InterventionType intervention);
```

Comment:
```suggestion
    bool is_compliant(RandomNumberGenerator& rng, InterventionType intervention) const;
```
Just noticed that this can now be marked const.

Reply:
Yes, thanks!